### ISSUE TITLE
Gem fails to compile on Windows

### ISSUE THREAD

**Issue by agross** [rubys/nokogumbo#9]
`gem install nokogumbo` yields:

```
Fetching: nokogumbo-1.1.9.gem (100%)
Temporarily enhancing PATH to include DevKit...
Building native extensions.  This could take a while...
ERROR:  Error installing nokogumbo:
        ERROR: Failed to build gem native extension.

    C:/Ruby/bin/ruby.exe extconf.rb
checking for xmlNewDoc() in -lxml2... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=C:/Ruby/bin/ruby
        --with-xml2lib
        --without-xml2lib
C:/Ruby/lib/ruby/2.0.0/mkmf.rb:434:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:519:in `try_link0'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:534:in `try_link'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:720:in `try_func'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:950:in `block in have_library'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:895:in `block in checking_for'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:340:in `block (2 levels) in postpone'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:310:in `open'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:340:in `block in postpone'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:310:in `open'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:336:in `postpone'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:894:in `checking_for'
        from C:/Ruby/lib/ruby/2.0.0/mkmf.rb:945:in `have_library'
        from extconf.rb:4:in `<main>'


Gem files will remain installed in C:/Ruby/lib/ruby/gems/2.0.0/gems/nokogumbo-1.1.9 for inspection.
Results logged to C:/Ruby/lib/ruby/gems/2.0.0/gems/nokogumbo-1.1.9/ext/nokogumboc/gem_make.out
```

mkmf.log:

```
"gcc -o conftest.exe -IC:/Ruby/include/ruby-2.0.0/i386-mingw32 -IC:/Ruby/include/ruby-2.0.0/ruby/backward -IC:/Ruby/include/ruby-2.0.0 -I. -DFD_SETSIZE=2048 -D_WIN32_WINNT=0x0501 -D_FILE_OFFSET_BITS=64   -O3 -fno-omit-frame-pointer -fno-fast-math -g -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wunused-variable -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -std=c99 conftest.c  -L. -LC:/Ruby/lib -L.      -lmsvcrt-ruby200  -lshell32 -lws2_32 -limagehlp -lshlwapi  "
In file included from C:/Ruby/include/ruby-2.0.0/ruby/defines.h:153:0,
                 from C:/Ruby/include/ruby-2.0.0/ruby/ruby.h:70,
                 from C:/Ruby/include/ruby-2.0.0/ruby.h:33,
                 from conftest.c:1:
C:/Ruby/include/ruby-2.0.0/ruby/win32.h: In function 'rb_w32_pow':
C:/Ruby/include/ruby-2.0.0/ruby/win32.h:801:5: warning: implicit declaration of function '_controlfp' [-Wimplicit-function-declaration]
C:/Ruby/include/ruby-2.0.0/ruby/win32.h:802:16: error: '_PC_64' undeclared (first use in this function)
C:/Ruby/include/ruby-2.0.0/ruby/win32.h:802:16: note: each undeclared identifier is reported only once for each function it appears in
C:/Ruby/include/ruby-2.0.0/ruby/win32.h:802:24: error: '_MCW_PC' undeclared (first use in this function)
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: #include <winsock2.h>
4: #include <windows.h>
5: int main(int argc, char **argv)
6: {
7:   return 0;
8: }
/* end */

```

**Comment by sono-it** [comment]
I'm having the exact same error as @agross, but instead with nokogumbo version 1.4.7 (ruby version is the same). I tried the fix suggested by @bodrovis, but it didn't seem to make a difference. 

Any suggestions? I can't find anything helpful anywhere else.

**Comment by stevecheckoway** [comment]
@danielrwolff I just pushed a new gem, a prerelease for 2.0.0. Can you try it on Windows and see if it works for you. I have it working in the Appveyor CI.

**Comment by ywjno** [comment]
@stevecheckoway Hi, in my PC for Windows 7, the `nokogumbo (2.0.0.pre.alpha)` is worked.

```
$ gem install nokogumbo --pre
Building native extensions.  This could take a while...
Successfully installed nokogumbo-2.0.0.pre.alpha
1 gem installed

$ ruby --version
ruby 2.2.5p319 (2016-04-26 revision 54774) [x64-mingw32]

$ irb
irb(main):001:0> require 'nokogumbo'
=> true
irb(main):002:0> Nokogumbo::VERSION
=> "2.0.0-alpha"
```

**Comment by stevecheckoway** [comment]
@ywjno Fantastic, thank you!

**Comment by stevecheckoway** [comment]
I'm going to mark this as closed now. We have Appveyor set up which should ensure that we keep builds on Windows working. If there are further issues, feel free to reopen this, or open a new issue.
